libc/src/stdio/baremetal/CMakeLists.txt

```diff
@@ -20,17 +43,27 @@ add_entrypoint_object(
     libc.include.stdio
 )
 
+add_entrypoint_object(
+  putc
+  SRCS
+    putc.cpp
+  HDRS
+    ../putc.h
+    write_utils.h
+  DEPENDS
+    libc.src.__support.CPP.string_view
+)
+
 add_entrypoint_object(
   printf
   SRCS
     printf.cpp
   HDRS
     ../printf.h
+    write_utils.h
```

Comment:
This shouldn't be here since it's not part of `printf` implementation, you should be instead depending on `baremetal_write_utils`.